pkg/controller/monitoring/monitoring_controller.go

```diff
@@ -0,0 +1,181 @@
+	}
+
+	if err = utils.AddPodMonitorWatch(c); err != nil {
+		return fmt.Errorf("monitoring-controller failed to watch PodMonitor resource: %w", err)
+	}
+
+	if err = utils.AddPrometheusRuleWatch(c); err != nil {
+		return fmt.Errorf("monitoring-controller failed to watch PrometheusRule resource: %w", err)
+	}
+
+	if err = utils.AddServiceMonitorCalicoNodeWatch(c); err != nil {
+		return fmt.Errorf("monitoring-controller failed to watch ServiceMonitor calico-node-monitor resource: %w", err)
+	}
+
+	if err = utils.AddServiceMonitorElasticsearchWatch(c); err != nil {
+		return fmt.Errorf("monitoring-controller failed to watch ServiceMonitor elasticsearch-metrics resource: %w", err)
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileMonitoring implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileMonitoring{}
+
+type ReconcileMonitoring struct {
+	client   client.Client
+	scheme   *runtime.Scheme
+	provider operatorv1.Provider
+	status   status.StatusManager
+}
+
+func getMonitoring(ctx context.Context, cli client.Client) (*operatorv1.Monitoring, error) {
+	instance := &operatorv1.Monitoring{}
+	err := cli.Get(ctx, utils.DefaultTSEEInstanceKey, instance)
+	if err != nil {
+		return nil, err
+	}
+
+	return instance, nil
+}
+
+func (r *ReconcileMonitoring) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Monitoring")
+
+	instance, err := getMonitoring(ctx, r.client)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			r.status.SetDegraded("An error occurred while querying Monitoring", err.Error())
+			return reconcile.Result{}, err
+		}
+		r.status.OnCRNotFound()
+	} else {
+		r.status.OnCRFound()
+	}
+
+	variant, installation, err := installation.GetInstallation(context.Background(), r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded("Installation not found", err.Error())
+			return reconcile.Result{}, err
+		}
+		r.status.SetDegraded("An error occurred while querying Installation", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// Create a component handler to manage the rendered component.
+	handler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
+
+	// render prometheus components
+	component := render.Monitoring(installation)
+
+	if err = imageset.ApplyImageSet(ctx, r.client, variant, component); err != nil {
+		reqLogger.Error(err, "Error with images from ImageSet")
+		r.status.SetDegraded("Error with images from ImageSet", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	if err := handler.CreateOrUpdateOrDelete(ctx, component, r.status); err != nil {
+		r.status.SetDegraded("Error creating / updating resource", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	r.status.ClearDegraded()
+
+	if instance != nil {
```

Comment:
It should never be possible for instance to be nil here. I think it is ok to leave this line as-is though.